lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
+ * @param {AppService} the AppService instance of the object
+ */
+Bridge.prototype.customiseAppservice = function(appService) {
+    // TODO(paul): This is gut-wrenching into the AppService instance itself.
+    //   Maybe an API on that object would be good?
+    var app = appService.app;
+
+    if (this.opts.controller.thirdPartyLookup) {
+        var lookupController = this.opts.controller.thirdPartyLookup;
+
+        var protocolFilter;
+        if ("protocols" in lookupController) {
+            protocolFilter = {};
+            lookupController.protocols.forEach(function (p) {
+                protocolFilter[p] = true;
+            });
+        }
+
+        app.get("/_matrix/app/:ver/thirdparty/protocol/:protocol", (req, res) => {
+            if (req.params.ver != "unstable") {
```

Comment:
Nit: Default to `!==` and only use `!=` if you have a good reason to please.

In this case it doesn't matter, both are equally (heh) good, but some linters will whine at you if you don't use triple =.